### ISSUE TITLE
Clean up auto formatting for yaml rules

### ISSUE THREAD

**Issue by ievans** [returntocorp/semgrep-rules#399]
It would be nice to automatically format our YAML files.

Desired features:
- [ ] Key ordering that is deterministic and consistent with our examples (eg `message`, `metadata` and `extra` are last)
- [ ] Keep the same syntax for `languages` (use `[js, python]` not indented newlines)
- [ ] Don't do extra quoting where it's not needed
- [ ] Trim trailing whitespace from fields like `id` and `message`

**Comment by ievans** [comment]
cc @rcoh who worked on this recently

**Comment by rcoh** [comment]
The yaml files are autoformatted with my fork of yamlfmt (https://github.com/rcoh/pre-commit-hook-yamlfmt), so it should be easy to make it do exactly what we want.

The thing to keep in mind is that semgrep.live needs to call an identical codepath or there will be precommit errors.

I tried to pick settings that were pretty close to what we had to make the diff small, but, eg. I explicitly disabled modifying quotes. We can disable that.
